pinot-common/src/main/java/org/apache/pinot/common/request/context/ExpressionContext.java

```diff
@@ -104,6 +104,19 @@ public void getColumns(Set<String> columns) {
     }
   }
 
+  public <T> T visit(ExpressionContextVisitor<T> visitor) {
```

Comment:
It is a little bit strange to me calling something a visitor without implementing the GoF visitor pattern, but I guess the problem here is the ExpressionContext class itself, which instead of being a sealed class with one subclass for each type, it is a single class with nullable attributes so... the classical GoF pattern cannot be applied as we use to.

Reply:
+1 I agree - I think if people like this approach I will refactor the ExpressionContext to follow the classical pattern

Reply:
i felt the same.